client/client.js

```diff
@@ -62,71 +65,119 @@ Client.create = function (message) {
 // Instance methods
 
 // Persist subscriptions and presences when not already persisted in memory
-Client.prototype.storeData = function (message) {
-  var subscriptions = this.subscriptions,
-      presences = this.presences,
-      changed = false,
-      lookupMessage;
+Client.prototype.storeData = function (messageIn) {
+  var subCount = Object.keys(this.subscriptions).length,
+      presCount = Object.keys(this.presences).length,
+      processedOp = false;
+
+  // Persist the message data, according to type
+  switch(messageIn.op) {
+    case 'unsubscribe':
+    case 'sync':
+    case 'subscribe':
+      processedOp = this._storeDataSubscriptions(messageIn);
+      break;
+
+    case 'set':
+      processedOp = this._storeDataPresences(messageIn);
+      break;
+  }
+
+  // FIXME: For now log everything Later, enable sample logging. 
+  // if (processedOp && subCount % LOG_WINDOW_SIZE === 0) {
+  if (processedOp) {
+    log.info('#storeData', { 
+      client_id: this.id,
+      subscription_count: subCount,
+      presence_count: presCount
+    });
+  }
+
+  return true;
+};
+
+Client.prototype._storeDataSubscriptions = function (messageIn) {
+  var message = _cloneForStorage(messageIn),
+      to = message.to,
+      subscriptionsKey = this.key + SUBSCRIPTIONS_SUFFIX,
+      existingSubscription;
 
   // Persist the message data, according to type
   switch(message.op) {
     case 'unsubscribe':
-      if (subscriptions[message.to]) {
-        delete subscriptions[message.to];
-        changed = true;
+      if (this.subscriptions[to]) {
+        delete this.subscriptions[to];
+        Core.Persistence.expire(subscriptionsKey, Client.getDataTTL());
+        Core.Persistence.deleteHash(subscriptionsKey, to);
+        return true;
       }
       break;
 
     case 'sync':
     case 'subscribe':
-      lookupMessage = subscriptions[message.to];
-      if (!lookupMessage ||
-            (lookupMessage.op != 'sync' && !_.isEqual(lookupMessage, message))) {
-        subscriptions[message.to] = message;
-        changed = true;
+      existingSubscription = this.subscriptions[to];
+      if (!existingSubscription ||
+            (existingSubscription.op != 'sync' && !_.isEqual(existingSubscription, message))) {
```

Comment:
I think `(existingSubscription.op != 'sync' && message.op == 'sync')` is a bit more clearer, but this is fine too.